fs-sim/src/System/FS/Sim/Prim.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UnboxedTuples              #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+module System.FS.Sim.Prim (
```

Comment:
I'm just wondering if this module deserves an explanation :thinking:

Reply:
I'll include it in a follow-up!